packages/replace/src/index.js

```diff
@@ -37,14 +37,22 @@ function mapToFunctions(object) {
 
 export default function replace(options = {}) {
   const filter = createFilter(options.include, options.exclude);
-  const { delimiters } = options;
+  const { delimiters, preventAssignment } = options;
   const functionValues = mapToFunctions(getReplacements(options));
-  const keys = Object.keys(functionValues)
-    .sort(longest)
-    .map(escape);
+  const keys = Object.keys(functionValues).sort(longest).map(escape);
+  if (![true, false].includes(preventAssignment)) {
```

Comment:
I think we'd rather warn if `preventAssignment` is `false|undefined`, bc the user needs to know that it'll be changing. Even if it is a bit noisy.

Reply:
Personally, I would prefer allowing the user to set their preference and not overriding it even if the default does change. As you say, it will be very noisy and there is no way to opt out of the warnings, even though (if the user has explicitly set the behaviour to false) there will be no breaking changes.

Reply:
Fair enough

Reply:
Given that, let's add a test to assert the warning is being thrown if not explicitly set. This is one of the plugins that are next up for TypeScript conversion, and I'd like to guard against regression there. e.g. https://github.com/rollup/plugins/blob/990ad2f0664460078ccca100be93daea2fbbbae4/packages/html/test/test.js#L63-L73